source/state_representation/include/state_representation/trajectory/CartesianTrajectory.hpp

```diff
@@ -0,0 +1,145 @@
+#include "state_representation/space/cartesian/CartesianState.hpp"
+#include "state_representation/trajectory/TrajectoryBase.hpp"
+
+namespace state_representation {
+
+/**
+ * @class CartesianTrajectoryPoint
+ * @brief Struct to represent a Cartesian trajectory point
+ */
+struct CartesianTrajectoryPoint : public TrajectoryPoint {};
+
+/**
+ * @class CartesianTrajectory
+ * @brief Class to represent a trajectory of Cartesian points and corresponding durations
+ */
+class CartesianTrajectory : public TrajectoryBase<CartesianTrajectoryPoint> {
+public:
+  /**
+   * @brief Empty constructor
+   */
+  explicit CartesianTrajectory();
+
+  /**
+   * @brief Constructor with name and reference frame provided
+   * @param name the name of the state
+   * @param reference_frame reference frame of the trajectory points
+   */
+  explicit CartesianTrajectory(const std::string& name, const std::string& reference_frame = "world");
+
+  /**
+   * @brief Constructor with initial point, duration, name, and reference frame provided
+   * @param point the initial point
+   * @param duration the initial duration
+   * @param name the name of the state
+   * @throw EmptyStateException if point is empty
+   * @throw IncompatibleReferenceFramesException if point has different reference frame
+   */
+  explicit CartesianTrajectory(
+      const std::string& name, const CartesianState& point, const std::chrono::nanoseconds& duration
+  );
+
+  /**
+   * @brief Constructor with intial points, durations, name and reference frame provided
+   * @param points vector of initial points
+   * @param durations vector of initial durations
+   * @param name the name of the state
+   * @throw EmptyStateException if any point is empty
+   * @throw IncompatibleReferenceFramesException if any point has different reference frame from others
+   * @throw IncompatibleSizeException if points and durations have different sizes
+   */
+  explicit CartesianTrajectory(
+      const std::string& name, const std::vector<CartesianState>& points,
+      const std::vector<std::chrono::nanoseconds>& durations
+  );
+
+  /**
+   * @brief Getter of the reference frame as const reference
+   * @return the reference frame associated with the trajectory
+   */
+  const std::string& get_reference_frame() const;
+
+  /**
+   * @brief Getter of the reference frame as const reference
+   * @param the reference frame associated with the trajectory
+   * @throws EmptyStateException if pose is empty
+   */
+  void set_reference_frame(const CartesianPose& pose);
+
+  /**
+   * @brief Add new point and corresponding duration to trajectory
+   * @param new_point the new trajectory point
+   * @param duration the duration for the new point
+   * @throw EmptyStateException if point is empty
+   * @throw IncompatibleReferenceFramesException if point has different reference frame
+   */
+  void add_point(const CartesianState& new_point, const std::chrono::nanoseconds& duration);
```

Comment:
nit: I would either go for `point` and `duration` or `new_point` and `new_duration` no? rather just `point`